lib/src/web/readers/configurations/emitter_configuration_reader.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+//
+// This program is licensed to you under the Apache License Version 2.0,
+// and you may not use this file except in compliance with the Apache License Version 2.0.
+// You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the Apache License Version 2.0 is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+import 'package:snowplow_tracker/configurations/emitter_configuration.dart';
+
+class EmitterConfigurationReader extends EmitterConfiguration {
+  EmitterConfigurationReader(dynamic map)
+      : super(serverAnonymisation: map['serverAnonymisation']);
+
+  void addTrackerOptions(dynamic options) {
+    var anonymousTracking = options.containsKey('anonymousTracking')
+        ? options['anonymousTracking']
+        : {};
+
+    if (serverAnonymisation == true) {
+      anonymousTracking['withServerAnonymisation'] = serverAnonymisation;
```

Comment:
I am wondering what happens if `TrackerConfiguration.sessionContext` is false, `TrackerConfiguration.userAnonymisation` is true, and `EmitterConfiguration.serverAnonymisation` is true. I think that in that case the `TrackerConfigurationReader` will assign the boolean `true` to `anonymousTracking` in `options` and this line will throw an error because it is trying to use the boolean as a dictionary.

I think we'll need to check here if `anonymousTracking` is a boolean and in that case replace it with a dictionary.